### PR TITLE
Global Block Theme Colors FE

### PR DESCRIPTION
**Pull request by ScottGarrison**
## What does this do/fix?

- Updates our theme classes to allow use for page and block level overrides.
- Global or Page level will add a `t-theme--light` or `t-theme--dark` class to the body. All blocks below will adhere to the selection
- Next level down at the Block, you can override the global or page, and the above classes get added to the parent wrapper of the block.
- If needed, above classes can be added closer to the element to keep overriding. 
- This all works based off custom properties and inheritance. In the future if a custom color picker is needed, a `style="--color-bgd-content: purple;"` can be used to override. Or create your own class to inject.

### CODE REVIEW

**Review by verytwisty — APPROVED** [review]

File: wp-content/themes/core/components/blocks/card_grid/Card_Grid_Controller.php [hunk 1]
```diff
@@ -76,7 +76,7 @@ protected function defaults(): array {
 	protected function required(): array {
 		return [
 			self::CONTAINER_CLASSES => [ 'b-card-grid__container', 'l-container' ],
-			self::LOOP_CLASSES      => [ 'b-card-grid__loop' ],
+			self::LOOP_CLASSES      => [ 'b-card-grid__loop', 't-theme--light' ],
```
Comment (dpellenwood) [inline wp-content/themes/core/components/blocks/card_grid/Card_Grid_Controller.php:79]: Curious why the `t-theme--light` class is necessary, especially as a required class. Is this the default anyway?

File: wp-content/themes/core/components/blocks/interstitial/Interstitial_Block_Controller.php [hunk 2]
```diff
@@ -136,8 +136,7 @@ public function get_content_args(): array {
 			Content_Block_Controller::CLASSES => [
 				'c-block__content-block',
 				'c-block__header',
-				'b-interstitial__content-container',
-				't-theme--light'
+				'b-interstitial__content-container'
```
Comment (dpellenwood) [inline wp-content/themes/core/components/blocks/interstitial/Interstitial_Block_Controller.php:139]: Please keep the trailing comma here.

File: wp-content/themes/core/components/blocks/spacer/css/spacer.pcss [hunk 3]
```diff
@@ -5,6 +5,8 @@
  * ----------------------------------------------------------------------------- */
 
 .b-spacer {
+	background: var(--color-bgd-content);
+
```
Comment (dpellenwood) [inline wp-content/themes/core/components/blocks/spacer/css/spacer.pcss:9]: Is the extra line break (before height) necessary?

File: wp-content/themes/core/components/card/css/card.pcss [hunk 4]
```diff
@@ -157,6 +159,12 @@
 	@mixin t-body-small;
 
 	color: var(--color-grey);
+
+	/* CASE: Dark Theme */
+	.t-theme--dark & {
+		color: var(--color-white);
+		opacity: 0.8;
```
Comment (dpellenwood) [inline wp-content/themes/core/components/card/css/card.pcss:166]: Why is the whole card's opacity set here? Wouldn't we only want to adjust the background color opacity?

**Review by dpellenwood — APPROVED** [review]
Couple questions and super minor changes, but nicely done, Scott!

**Comment by defunctl** [comment]
@dpellenwood I am going to merge this into my branch, we'll still have a chance to review the entire feature with my PR

**Pull request merged**